internal/provider/atlas_migration_resource.go

```diff
@@ -376,42 +389,140 @@ func (r *MigrationResource) migrate(ctx context.Context, data *MigrationResource
 	}
 	defer os.RemoveAll(dir)
 	cfgPath := filepath.Join(dir, "atlas.hcl")
+	if data.Cloud != nil && data.Cloud.Token.ValueString() != "" {
+		// Override cloud config from provider if cloud config from resource is set
+		r.cloud = data.Cloud
+	}
 	err = data.AtlasHCL(cfgPath, r.devURL, r.cloud)
 	if err != nil {
 		diags.AddError("Generate config failure",
 			fmt.Sprintf("Failed to create atlas.hcl: %s", err.Error()))
 		return
 	}
-	statusReport, err := r.client.MigrateStatus(ctx, &atlas.MigrateStatusParams{
-		ConfigURL: fmt.Sprintf("file://%s", filepath.ToSlash(cfgPath)),
-		Env:       defaultString(data.EnvName, "tf"),
-	})
-	if err != nil {
-		diags.AddError("Failed to read migration status", err.Error())
-		return
-	}
-	amount, synced := statusReport.Amount(data.Version.ValueString())
-	if !synced {
-		if amount == 0 {
-			diags.AddAttributeError(
-				path.Root("version"),
-				"Incorrect version",
-				"The version is not found in the pending migrations.",
-			)
+	var status *atlas.MigrateStatus
+	switch {
+	case r.cloud != nil && !data.RemoteDir.Name.IsNull():
+		status, err = r.client.MigrateStatus(ctx, &atlas.MigrateStatusParams{
+			ConfigURL: fmt.Sprintf("file://%s", filepath.ToSlash(cfgPath)),
+			Env:       defaultString(data.EnvName, "tf"),
+		})
+		if err != nil {
+			diags.AddError("Failed to read migration status", err.Error())
 			return
 		}
-		_, err := r.client.MigrateApply(ctx, &atlas.MigrateApplyParams{
+	default:
+		dirURL, err := url.Parse(data.DirURL.ValueString())
```

Comment:
Could this potentially be a bug, considering it's user input from different operating systems? Should I use `filepath.Fromslash` here? @giautm